explore/src/main/scala/explore/cache/ModesCache.scala

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.cache
+
+import cats.effect.IO
+import cats.effect.kernel.Resource
+import cats.syntax.all.*
+import clue.StreamingClient
+import explore.DefaultErrorPolicy
+import explore.modes.SpectroscopyModesMatrix
+import fs2.Stream
+import japgolly.scalajs.react.*
+import lucuma.react.common.ReactFnProps
+import lucuma.schemas.ObservationDB
+import queries.common.ModesQueriesGQL
+
+case class ModesCache(
```

Comment:
I wonder if this is overkill for this case

Reply:
Do you mean caching the modes might be overkill?

Reply:
We want to call it only once not  every time we open the modees tile, in that sense caching is important
However they also don't (often) change upstream, there is no subscription for example

Reply:
We'll have to figure this out when there is filtering on the ODB for CfP, resource availability, etc. But for now this is 👍